src/relayFeeCalculator/relayFeeCalculator.ts

```diff
@@ -61,19 +61,11 @@ export class RelayFeeCalculator {
 
   constructor(config: RelayFeeCalculatorConfig, logger: Logger = DEFAULT_LOGGER) {
     this.queries = config.queries;
-    this.gasDiscountPercent = config.gasDiscountPercent || 0;
-    this.capitalDiscountPercent = config.capitalDiscountPercent || 0;
+    this.gasDiscountPercent = config.gasDiscountPercent || {};
+    this.capitalDiscountPercent = config.capitalDiscountPercent || {};
```

Comment:
Does it make sense to discount the relayer fees at all?

The benefit of per-destination pricing is that we can reduce the burden on HubPool funds. The relayer itself is unaffected by this - its costs are still the same, and it is still without its funds until the next bundle is executed.

I think it's more logical to me to apply a discount to the LP fee, rather than the relayer fee.

Reply:
Yeah, +1 to this question. I'm also curious about the motivation.

Reply:
Oh wow I hadn't thought about this. So an LP fee discount could be implemented by adding a key to the `l1TokenConfig` state in the [`AcrossConfigStore`](https://etherscan.io/address/0x3b03509645713718b78951126e0a6de6f10043f5#readContract) like:

`lpFeeDiscount: { [chainId]: number >= 0 && number <= 100 }`

And then the relayer could take the lpFeeDiscount for the token and destination chain and apply it to the lpFee [here](https://github.com/across-protocol/relayer-v2/blob/master/src/clients/ConfigStoreClient.ts#L65) in the `ConfigStoreClient`

Thoughts on this design?

Reply:
Storing a discount on-chain will mean that we need to be pretty conservative in applying any discounts, since we won't be reflexive to changes in demand with a static discount. We risk ending up with a huge flood of funds rushing to L2s, and by the time we do react to it, we might have a high HubPool utilisation and a poor LP APY.

I think a nice solution would factor in the SpokePool balance on the destination chain in order to determine whether it's above or below a target balance, and apply an LP fee modifier accordingly. This is probably deceptively complex though, and there are definitely race conditions that might lead to unexpected outcomes.

As a meta question: When we are modifying the LP fee derivation, how do we ensure that it is rolled out amongst all relayers? I checked with one third-party relayer and they are using the SDK, so unless they synchronise an update of their SDK, they will end up making fills with LP fees that we disagree on. We also have various aggregators that are generating deposit transactions; we assume that they are querying the API suggested-fees endpoint, but it's hard to know for sure...

Reply:
I think we should deprecate this PR in favor of https://github.com/across-protocol/relayer-v2/pull/391 which is more flexible than discounting the default LP fee. We can set a per route rate model.

Regarding your meta-question: 

>"As a meta question: When we are modifying the LP fee derivation, how do we ensure that it is rolled out amongst all relayers?"

I think that this SDK version bump is the best way to roll this out, so we should make sure to implement the change in #391 change (along with a corresponding SDK change) to be backwards compatible and give time for other relayers to upgrade their versions. The `routeRateModels` won't be added initially so we will give relayers plenty of time to upgrade before the `routeRateModels` start affecting the LP fees